lib/requirements/form/content_checker.rb

```diff
@@ -6,13 +6,9 @@ def initialize(edition, params)
     @params = params
   end
 
-  def issues
-    issues = Requirements::CheckerIssues.new
-
+  def check
     edition.document_type.contents.each do |field|
-      issues += field.pre_update_issues(edition, params)
+      self.issues += field.form_issues(edition, params)
```

Comment:
This seemed a little surprising. How would you feel about a merge method on issues that combined a collection? I assume we don't ever need to actually reset the issues to an empty collection.

Reply:
Yes, it was a bit annoying to have to use `self`. I did try out a `merge!` method on CheckerIssues, so the above line would become

```
issues.merge!(field.form_issues(edition, params))
```

Is that what you were thinking?

There wasn't much in it for me: not using a mutable `!` method seemed clearer, and it meant I didn't need to modify the CheckerIssues class.

Reply:
Yup - just like that but without the bang (since there isn't a non-bang merge method). 

I think it'd be worth the small change to resolve the subtle wtfs the assignment approach caused (for example, I was quite surprised we had an attr_accessor)

Reply:
I'd like to suggest an alternative: `push`. The reason I avoided `merge` is that I'm used to its immutable behaviour for hashes, so although there's only one method, it would be surprising to me. Whereas `push` already has mutable semantics for `Array`.

Any particular concern with using `push`?

Reply:
I'm happy with push 👍

Reply:
I've added a commit for that: https://github.com/alphagov/content-publisher/pull/2017/commits/0ae2c8883b973a77dbd265dc5c8ecd348f2e3959. Can you take a look?